main/maven/test/src/mill/main/maven/BuildGenTests.scala

```diff
@@ -125,15 +131,78 @@ object BuildGenTests extends TestSuite {
       .map(PathRef(_))
       .toSeq
 
-  def checkFiles(actualFiles: Seq[PathRef], expectedRoot: os.Path): Boolean = {
-    val expectedFiles = buildFiles(expectedRoot)
+  def checkFiles(actualFiles: Seq[os.SubPath], root: os.Path, expectedRoot: os.Path): Boolean = {
```

Comment:
Could we use `git diff --no-index` to perform this comparison? That comes with a nice diff viewer for free for people to understand what the difference between the two folders is, and means a lot less comparison code for us to maintain

Reply:
Trying that. That output is mainly meant for CI IMO. Locally, one can set `updateSnapshots` to true, and run `git diff` manually to check the diff.